### PR TITLE
checker: refactor fixOrphanPeers

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref #4399

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -475,16 +483,15 @@ func (c *RuleChecker) fixOrphanPeers(region *core.RegionInfo, fit *placement.Reg
...
-				// make sure is down peer.
-				if region.GetDownPeer(p.GetId()) != nil {
-					return p, true
-				}
-				return nil, true
+			if isInDisconnectedStore(p) {
+				return p, true
 			}
-			if isDisconnectedPeer(p) {
+			if isDownPeer(p.GetId()) {
```
Comment (rleungx) [inline pkg/schedule/checker/rule_checker.go:489]: How about using `||`?

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Comment by rleungx** [comment]
ci failed

**Pull request merged**